modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -171,8 +173,18 @@ public CompletableFuture<Void> notifyWatches(List<Entry> updatedEntries, HybridT
                     // Revision must be the same for all entries.
                     long newRevision = updatedEntries.get(0).revision();
 
+                    List<Entry> filteredUpdatedEntries = updatedEntries.stream()
+                            .filter(entry ->
+                                    entry.key().length <= IDEMPOTENT_COMMAND_PREFIX_BYTES.length
+                                            ||
+                                            entry.key().length > IDEMPOTENT_COMMAND_PREFIX_BYTES.length
+                                                    && !ByteBuffer.wrap(entry.key(), 0, IDEMPOTENT_COMMAND_PREFIX_BYTES.length)
+                                                            .equals(ByteBuffer.wrap(IDEMPOTENT_COMMAND_PREFIX_BYTES)))
+                            .collect(Collectors.toList());
+
```

Comment:
Is this necessary? I mean, if there are no subscriptions on IDEMPOTENT_COMMAND_PREFIX_BYTES these entries would be just ignored, right?

Reply:
Yep, there are cases when we subscribe to all keys, definitely in some tests.